operator/gitops/argocd/pipeline-service/openshift-pipelines/appstudio-pipelines-runner-clusterrole.yaml

```diff
@@ -0,0 +1,49 @@
+---
+# Any change to this role must be communicated to the devsandbox team
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: appstudio-pipelines-runner-clusterrole
+rules:
+  - verbs:
+      - use
+    apiGroups:
+      - security.openshift.io
+    resources:
+      - securitycontextconstraints
+    resourceNames:
+      - pipelines-scc
+  - verbs:
+      - get
+      - list
+      - watch
+    apiGroups:
+      - ''
+    resources:
+      - configmaps
+      - secrets
+  - verbs:
+      - create
+      - get
+      - list
+      - patch
+      - update
+      - watch
+    apiGroups:
+      - ''
+      - image.openshift.io
+    resources:
+      - imagestreamimages
+      - imagestreammappings
+      - imagestreams
+      - imagestreams/secrets
+      - imagestreamtags
+      - imagetags
+  - verbs:
+      - get
+      - update
+    apiGroups:
+      - ''
+      - image.openshift.io
+    resources:
+      - imagestreams/layers
```

Comment:
We use imagestreams in app studio? Do we also use the internal registry? 

If so, @gabemontero is this the right set of permissions needed to push to the internal registry?

Reply:
I am not aware of usage of imagestreams

Reply:
yeah way back when, like back in 2022 when I first joined appstudio, I remember there being some options to use imagestreams as a destination for builds in the build-definitions tests or something like that, but I don't see it in that repo anymore

and certainly I would consider @Michkov the source of truth wrt along with @sbose78 for use *within* app studio.

That said, if a customer can provide their own pipelines via PaC, in theory they could want to use imagestreams.  The openshift-pipelines team I believe done work to facilitate that, with the `pipeline` service account

Why do we need both this new service account and the openshift-pipelines `pipeline` SA ?

Pending the answers to those questions, if these permissions are needed, instead of what is done here, openshift creates ClusterRoles for poth pulling and pushing images from/to the internal image registry, and in fact, whenever a namespace or project is created, RoleBindings for those ClusterRoles are created for imagestreams defined in that namepsace and associated with predefined SA's like the openshift `builder` SA.

For example:

```
gmontero ~ $ oc get rolebinding system:image-pullers -o yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  annotations:
    openshift.io/description: Allows all pods in this namespace to pull images from
      this namespace.  It is auto-managed by a controller; remove subjects to disable.
  creationTimestamp: "2023-05-02T16:05:42Z"
  name: system:image-pullers
  namespace: ggmtest
  resourceVersion: "90370"
  uid: bf7efe78-5f35-4e06-820e-084ba40b496d
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: system:image-puller
subjects:
- apiGroup: rbac.authorization.k8s.io
  kind: Group
  name: system:serviceaccounts:ggmtest
gmontero ~ $ oc get rolebinding system:image-builders -o yaml 
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  annotations:
    openshift.io/description: Allows builds in this namespace to push images to this
      namespace.  It is auto-managed by a controller; remove subjects to disable.
  creationTimestamp: "2023-05-02T16:06:48Z"
  name: system:image-builders
  namespace: ggmtest2
  resourceVersion: "90862"
  uid: 9e85e373-ba7b-4d21-a457-9963854f089d
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: system:image-builder
subjects:
- kind: ServiceAccount
  name: builder
  namespace: ggmtest2
gmontero ~ $ 
```

So @Roming22 you should just reuse the existing `system:image-builders` and `system:image-pullers` ClusterRoles and create RoleBindings and/or ClusterRoleBindings as needed to allow interaction with imagestreams and the internal image registry.

Reply:
The reason behind these permissions is that they are used by the pipeline-service tests. The test for tekton-chains will copy an image to the internal registry and make sure that the attestation has been created.

I'll remove those from the "staging/production" configuration, and add it as part of the "development" configuration.

Reply:
Fixed.